p03-form-validations/src/Form/Form.component.js

```diff
@@ -1,20 +1,75 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
 import { Card } from '../Shared/components/Card'
 import { Button, Input } from '../Shared/components'
-import { fieldsList } from './Form.services'
+import { fieldsList, isFieldValid } from './Form.services'
 
 import "./Form.styles.css"
 
+const initialState = fieldsList.map(field=> ({...field, ready:false, value:""}))
+
 const Form = () => {
-  
+
+  const [fieldsReady, setFieldsReady] = useState(false)
+  const [listo, setListo] = useState('Not ready to send')
```

Comment:
The name of the state variables may be more expressive in their intention, in that way when our future self or a coworker read the code, can be easily understood.

in this case this could be "btnSubscribeTitle" or just "subscribeTitle". 

by the way, this flow of reassigning a button title it's perfectly fine, but I just going to show you how you can do it differently on the button itself.